client/landing/stepper/declarative-flow/internals/steps-repository/playground/lib/blueprint.ts

```diff
@@ -105,14 +105,35 @@ function getDefaultBlueprint( recommendedPhpVersion: string ): Blueprint {
 	};
 }
 
-export function getBlueprintName( name: string | null ): string | null {
+function getBlueprintName( name: string | null ): string | null {
 	if ( name && name in PREDEFINED_BLUEPRINTS ) {
 		return name;
 	}
 
 	return null;
 }
 
+// Used in sending the Tracks event
+export function getBlueprintLabelForTracking( query: URLSearchParams ): string {
+	const name = query.get( 'blueprint' );
+
+	if ( name && name in PREDEFINED_BLUEPRINTS ) {
+		return name;
+	}
+
+	// If its a blueprintlibrary.wordpress.com url for blueprint, use its id to construct the label
+	const blueprintUrl = query.get( 'blueprint-url' );
+	if ( blueprintUrl ) {
+		const src = new URL( blueprintUrl );
+		if ( src.host === 'blueprintlibrary.wordpress.com' ) {
```

Comment:
Maybe creating a new constant for the Blueprint Library URL is a good idea, to be used somewhere else. What do you think?

Reply:
Added!